Services/Database/classes/Setup/class.ilMysqlMyIsamToInnoDbMigration.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace ILIAS\Setup;
+
+use ilDatabaseInitializedObjective;
+use ilDatabaseUpdatedObjective;
+use ilDBConstants;
+use ilException;
+use ILIAS\Setup;
+use ilIniFilesLoadedObjective;
+
+class ilMysqlMyIsamToInnoDbMigration implements Migration
+{
+    protected ?string $db_name = null;
+    protected ?\ilDBInterface $database = null;
+
+    /**
+     * @inheritDoc
+     */
+    public function getLabel() : string
+    {
+        return "Migration to convert tables from MyISAM to Innodb service";
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getDefaultAmountOfStepsPerRun() : int
+    {
+        return 20;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPreconditions(Environment $environment) : array
+    {
+        return [
+            new ilIniFilesLoadedObjective(),
+            new ilDatabaseInitializedObjective(),
+            new ilDatabaseUpdatedObjective()
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function prepare(Environment $environment) : void
+    {
+        /**
+         * @var $client_id  string
+         */
+        $this->database = $environment->getResource(Environment::RESOURCE_DATABASE);
+        $client_ini = $environment->getResource(Environment::RESOURCE_CLIENT_INI);
+        $this->db_name = $client_ini->readVariable('db', 'name');
+    }
+
+    /**
+     * @inheritDoc
+     * @throws ilException
+     */
+    public function step(Environment $environment) : void
+    {
+        $errors = $this->database->migrateAllTablesToEngine();
```

Comment:
we should migrate one by one here, maybe we introduce another method for that in `ilDBInterface` for this

Reply:
@chfsx 
Then we would need a new function in the DBInterface as well as in the other classes, should I implement this, or will you implement this functions?

Reply:
@chfsx 
Thx! And done.